libs/dh/profile/feature-profile-modal/src/lib/dh-profile-modal/dh-profile-modal.component.ts

```diff
@@ -0,0 +1,136 @@
+import { WattToastService } from '@energinet-datahub/watt/toast';
+import { VaterFlexComponent, VaterStackComponent } from '@energinet-datahub/watt/vater';
+import { TranslocoDirective, translate } from '@ngneat/transloco';
+
+type UserPreferencesForm = FormGroup<{
+  email: FormControl<string>;
+  phone: FormControl<string>;
+  language: FormControl<string>;
+  firstname: FormControl<string>;
+  lastname: FormControl<string>;
+}>;
+
+@Component({
+  selector: 'dh-profile-modal',
+  standalone: true,
+  styles: `.names {
+    width:75%;
+  }`,
+  imports: [
+    WATT_MODAL,
+    WattTextFieldComponent,
+    WattPhoneFieldComponent,
+    WattDropdownComponent,
+    WattButtonComponent,
+    TranslocoDirective,
+    VaterStackComponent,
+    VaterFlexComponent,
+    DhDropdownTranslatorDirective,
+    ReactiveFormsModule,
+  ],
+  template: `<watt-modal *transloco="let t; read: 'shared.profile'" [title]="t('title')">
+    <form [formGroup]="userPreferencesForm" id="userPreferencesForm" (ngSubmit)="save()">
+      <vater-flex fill="vertical" gap="m">
+        <vater-stack class="names" align="flex-start" direction="column" gap="s">
+          <h4>{{ t('myInformation') }}</h4>
+          <vater-stack align="flex-start" direction="row" gap="s">
+            <watt-text-field
+              [label]="t('name')"
+              [formControl]="userPreferencesForm.controls.firstname"
+            />
+            <watt-text-field
+              [label]="t('lastname')"
+              [formControl]="userPreferencesForm.controls.lastname"
+            />
+          </vater-stack>
+
+          <watt-text-field
+            [label]="t('email')"
+            [formControl]="userPreferencesForm.controls.email"
+          />
+          <watt-phone-field
+            [label]="t('phone')"
+            [formControl]="userPreferencesForm.controls.phone"
+          />
+          <h4>{{ t('general') }}</h4>
+          <watt-dropdown
+            [label]="t('language')"
+            [formControl]="userPreferencesForm.controls.language"
+            [options]="languages"
+            dhDropdownTranslator
+            translate="shared.profile.languages"
+          />
+        </vater-stack>
+      </vater-flex>
+    </form>
+    <watt-modal-actions>
+      <watt-button variant="secondary" (click)="closeModal(false)">{{ t('cancel') }}</watt-button>
+      <watt-button variant="secondary" type="submit" formId="userPreferencesForm">{{
+        t('save')
+      }}</watt-button>
+    </watt-modal-actions>
+  </watt-modal>`,
+})
+export class DhProfileModalComponent {
+  private readonly _formBuilder = inject(NonNullableFormBuilder);
+  private readonly _toastService = inject(WattToastService);
+
+  @ViewChild(WattModalComponent)
+  private _profileModal!: WattModalComponent;
+
+  languages: WattDropdownOptions = [
+    { value: 'en', displayValue: 'English' },
+    { value: 'da', displayValue: 'Danish' },
+  ];
```

Comment:
suggestion: consider using `DisplayLanguage` as a type/value for the `language` control.